### PR TITLE
Charles rails validations

### PR DESCRIPTION
**Pull request by charlesx2013**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
+[![Code Climate](https://codeclimate.com/github/calblueprint/watershed/badges/gpa.svg)](https://codeclimate.com/github/calblueprint/watershed)
+
```
Comment (nishdesai) [inline README.md]: can you move this to the bottom, next to the codeship build?

File: md [hunk 2]
```diff
@@ -0,0 +1 @@
+[?1049h[1;25r(B[m[4l[?7h[?12l[?25h[?1h=[?1h=[?1h=[39;49m[39;49m(B[m[H[2J(B[0;7m  GNU nano 2.0.6                                    File: README                                                                                [23;67H[ New File ][24d^G(B[m Get Help[24;25H(B[0;7m^O(B[m WriteOut[24;49H(B[0;7m^R(B[m Read File[24;73H(B[0;7m^Y(B[m Prev Page[24;97H(B[0;7m^K(B[m Cut Text[24;121H(B[0;7m^C(B[m Cur Pos[25d(B[0;7m^X(B[m Exit[25;25H(B[0;7m^J(B[m Justify[25;49H(B[0;7m^W(B[m Where Is[25;73H(B[0;7m^V(B[m Next Page[25;97H(B[0;7m^U(B[m UnCut Text[25;121H(B[0;7m^T(B[m To Spell[3d[24d[J[25;144H[25;1H[?1049l[?1l>
```
Comment (nishdesai) [inline md]: what's this? I think it can go.

**Comment by nishdesai** [comment]
@charlesx2013 can you give a summary of the features implemented in this pr?

**Comment by charlesx2013** [comment]
added backend validations and added a claim feature for tasks

**Pull request merged**